### PR TITLE
Final iteration of performance tweaks

### PR DESCRIPTION
**Pull request by mrmlnc**
## Proposed Changes

Fix for #26.

### Benchmark [only 5x loop]

| before | after    | faster |
|--------|----------|--------|
| 8 op/s | 111 op/s | 13.8x  |

### Benchmark [complex from [posthtml-benchmark](https://github.com/mrmlnc/posthtml-benchmark)]

| before | after   | faster |
|--------|---------|--------|
| 7 op/s | 61 op/s | 8.7x   |

## Types of Changes

Fully compatible with the current version.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -257,28 +273,54 @@ function parseLoopStatement (input) {
...
 
 /**
+ * Creates a backup of keys values
+ */
+function makeLocalsBackup (keys, locals) {
+  let backup = {}
+
+  for (let i = 0; i < keys.length; i++) {
+    const key = keys[i]
+    if (locals.hasOwnProperty(key)) backup[key] = cloneDeep(locals[key])
```
Comment (voischev) [inline lib/index.js]: maybe `backup[key] = Object.create(locals[key])` ??

Comment (voischev) [inline lib/index.js]: or JSON.stringify+JSON.parse? :D

Comment (mrmlnc) [inline lib/index.js]: > maybe backup[key] = Object.create(locals[key]) ??

But if:

```js
locals[key] = {
  nested: {
    a: [1, 2, 3]
  }
}
```

Comment (mrmlnc) [inline lib/index.js]: Ooops,

image

Comment (voischev) [inline lib/index.js]: 👍

**Comment by voischev** [comment]
@mrmlnc thx for great PR!

**Comment by michael-ciniawsky** [comment]
@mrmlnc You're are also admin here now, sry for the delay I forgot to do it recently

**Comment by mrmlnc** [comment]
@michael-ciniawsky,

👍

**Comment by michael-ciniawsky** [comment]
@mrmlnc I update at least the README (already in progress (90%)) and make new release. Afterwards I add you as npm owner aswell. (we need a npm release bot for posthtml org 😛, but off-topic)

**Comment by mrmlnc** [comment]
@michael-ciniawsky,

I think is better to not give me the rights for plugins in npm, because I prefer to create PR 😺

**Comment by michael-ciniawsky** [comment]
Doesn't hurt for asap bug fixes 😛 , e.g in case i'm busy with social work etc., making PR for features is good practice 👍 nevertheless + do as you prefer 😎

**Pull request merged**